src/uct/cuda/cuda_ipc/cuda_ipc_cache.c

```diff
@@ -200,41 +234,96 @@ uct_cuda_ipc_open_memhandle_vmm(uct_cuda_ipc_rkey_t *key,
     return status;
 }
 
-static ucs_status_t
-uct_cuda_ipc_open_memhandle_mempool(uct_cuda_ipc_rkey_t *key,
-                                    CUdeviceptr *mapped_addr)
+static ucs_status_t cuda_ipc_rem_mpool_cache_create(uct_cuda_ipc_rkey_t *key,
+                                                    CUmemoryPool *mpool,
+                                                    CUdeviceptr *mapped_addr)
 {
     CUmemAccessDesc access_desc = {};
-    CUdeviceptr dptr;
     ucs_status_t status;
 
     status = UCT_CUDADRV_FUNC_LOG_ERR(cuMemPoolImportFromShareableHandle(
-                &key->ph.pool, (void *)&key->ph.handle.fabric_handle,
+                mpool, (void *)&key->ph.handle.fabric_handle,
                 CU_MEM_HANDLE_TYPE_FABRIC, 0));
     if (status != UCS_OK) {
-        return status;
+        goto err;
     }
 
-    status = UCT_CUDADRV_FUNC_LOG_ERR(cuMemPoolImportPointer(&dptr,
-                key->ph.pool, (CUmemPoolPtrExportData*)&key->ph.ptr));
+    status = UCT_CUDADRV_FUNC_LOG_ERR(cuMemPoolImportPointer(mapped_addr,
+                *mpool, (CUmemPoolPtrExportData*)&key->ph.ptr));
     if (status != UCS_OK) {
-        return status;
+        goto err;
     }
 
     status = uct_cuda_ipc_init_access_desc(&access_desc);
     if (status != UCS_OK) {
-        return status;
+        goto err;
     }
 
-    status = UCT_CUDADRV_FUNC_LOG_ERR(cuMemPoolSetAccess(key->ph.pool,
-                &access_desc, 1));
+    status = UCT_CUDADRV_FUNC_LOG_ERR(
+                cuMemPoolSetAccess(*mpool, &access_desc, 1));
     if (status != UCS_OK) {
-        return status;
+        goto err;
     }
 
-    *mapped_addr = dptr;
-
     return UCS_OK;
+
+err:
+    return status;
```

Comment:
why replaced "return status" by "goto err"?
do we need revert the import operation of a subsequent API call (such as cuMemPoolSetAccess) fails?

Reply:
I think it makes sense to call `cuMemPoolDestroy(mpool)` before `goto err` as per @yosefe's suggestion. @iyastreb would you mind making the change?

Reply:
np, I'll do it

Reply:
I added a bit more than that:
```C
err_free_ptr:
    cuMemFree(dptr);   << Also destroy imported memory
err_free_mpool:
    cuMemPoolDestroy(*mpool);
    return status;
```
As per documentation of `cuMemPoolImportPointer `:
> The imported memory must not be accessed before the allocation operation completes in the exporting process. The imported memory must be freed from all importing processes before being freed in the exporting process. The pointer may be freed with cuMemFree or cuMemFreeAsync.